src/common/github/__tests__/index.ts

```diff
@@ -198,6 +198,173 @@ describe('GitHub utils', () => {
     })
   })
 
+  describe('UncomittedChangesWorkflow', () => {
+    const workflowPath = '.github/workflows/uncomitted-changes.yml'
+
+    test('throws if called with a project not derived from NodeProject', () => {
+      const project = new JavaProject({name: 'java', groupId: 'java', artifactId: 'app', version: '0', github: true})
+      expect(() => new PullRequestTest(project.github!)).toThrow()
+    })
+
+    test('creates a workflow', () => {
+      const project = new TestProject()
+      new UncomittedChangesWorkflow(project.github!)
+      const snapshot = synthSnapshot(project)
+      expect(snapshot).toMatchSnapshot()
+      expect(snapshot[workflowPath]).toBeDefined()
+    })
+
+    test('configures the workflow to be run on pull requests and pushes to main branch with changes to projenrc file', () => {
+      const project = new TestProject()
+      const paths = ['.projenrc.js']
+      new UncomittedChangesWorkflow(project.github!)
+      const snapshot = synthSnapshot(project)
+      const workflow = YAML.parse(snapshot[workflowPath])
+
+      expect(workflow.on).toEqual({
+        pull_request: {paths, types: ['opened', 'synchronize']},
+        push: {paths, branches: ['main']},
+      })
+    })
+
+    test('allows to be run on pushes to specified branches', () => {
+      const project = new TestProject()
+      const branches = ['main', 'dev']
+      const paths = ['.projenrc.js']
+      new UncomittedChangesWorkflow(project.github!, {triggerOnPushToBranches: branches})
+      const snapshot = synthSnapshot(project)
+      const workflow = YAML.parse(snapshot[workflowPath])
+
+      expect(workflow.on).toEqual({
+        pull_request: {paths, types: ['opened', 'synchronize']},
+        push: {paths, branches},
+      })
+    })
+
+    test('adds a check for uncomitted changes to the workflow', () => {
+      const project = new TestProject()
+      new UncomittedChangesWorkflow(project.github!)
+      const snapshot = synthSnapshot(project)
+      const workflow = YAML.parse(snapshot[workflowPath])
+      const jobs = Object.values<projen.github.workflows.Job>(workflow.jobs).map((j) => j.steps.at(-1)!.name)
+      expect(jobs).toHaveLength(1)
+      expect(jobs[0]).toEqual('Check git')
+    })
+
+    test('allows setting pnpm as a package manager', () => {
```

Comment:
@quesabe do we really need some of these cases?

Reply:
They just make sure it works with different setups. However this functionality comes from the `setupNode` function. So, your are right, we can remove it here.